src/main/java/pl/cleankod/exchange/entrypoint/ExceptionHandlerAdvice.java

```diff
@@ -5,15 +5,36 @@
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import pl.cleankod.exchange.core.usecase.CurrencyConversionException;
 import pl.cleankod.exchange.entrypoint.model.ApiError;
+import pl.cleankod.exchange.provider.nbp.exceptions.NbpAPIException;
 
 @ControllerAdvice
 public class ExceptionHandlerAdvice {
 
     @ExceptionHandler({
             CurrencyConversionException.class,
-            IllegalArgumentException.class
     })
     protected ResponseEntity<ApiError> handleBadRequest(CurrencyConversionException ex) {
         return ResponseEntity.badRequest().body(new ApiError(ex.getMessage()));
     }
+
+    @ExceptionHandler({
+            IllegalArgumentException.class
+    })
+    protected ResponseEntity<ApiError> handleIllegalArgumentExcheption(IllegalArgumentException ex) {
+        return ResponseEntity.badRequest().body(new ApiError("Please check the provided arguments"));
```

Comment:
As an API consumer I would not be very happy with such a response. It does not give me any clue what I did wrong. How can you improve on this?